pex/resolve/locker.py

```diff
@@ -359,14 +362,15 @@ def analyze(self, line):
                 elif isinstance(artifact_url.scheme, ArchiveScheme.Value):
                     selected_path = os.path.basename(artifact_url.path)
                     source_archive_path = os.path.join(self._download_dir, selected_path)
-                    if not os.path.isfile(source_archive_path):
-                        raise AnalyzeError(
-                            "Failed to lock {artifact}. Could not obtain its content for "
-                            "analysis.".format(artifact=artifact_url)
-                        )
-                    digest = Sha256()
-                    hashing.file_hash(source_archive_path, digest)
-                    source_fingerprint = Fingerprint.from_digest(digest)
+                    # If Pip resolves the artifact from its own cache, we will not find it in the
```

Comment:
Prior to this fix the new test fails with:
```
$ tox -epy311-integration -- -vvsk test_issue_2098 -n0
...
E       AssertionError: integration test failed: return_code=1, output=, error=running sdist
E       running egg_info
E       creating my_feast.egg-info
E       writing my_feast.egg-info/PKG-INFO
E       writing dependency_links to my_feast.egg-info/dependency_links.txt
E       writing requirements to my_feast.egg-info/requires.txt
E       writing top-level names to my_feast.egg-info/top_level.txt
E       writing manifest file 'my_feast.egg-info/SOURCES.txt'
E       reading manifest file 'my_feast.egg-info/SOURCES.txt'
E       writing manifest file 'my_feast.egg-info/SOURCES.txt'
E       running check
E       creating my_feast-0.0.1
E       creating my_feast-0.0.1/my_feast.egg-info
E       copying files to my_feast-0.0.1...
E       copying README.rst -> my_feast-0.0.1
E       copying pyproject.toml -> my_feast-0.0.1
E       copying my_feast.egg-info/PKG-INFO -> my_feast-0.0.1/my_feast.egg-info
E       copying my_feast.egg-info/SOURCES.txt -> my_feast-0.0.1/my_feast.egg-info
E       copying my_feast.egg-info/dependency_links.txt -> my_feast-0.0.1/my_feast.egg-info
E       copying my_feast.egg-info/requires.txt -> my_feast-0.0.1/my_feast.egg-info
E       copying my_feast.egg-info/top_level.txt -> my_feast-0.0.1/my_feast.egg-info
E       Writing my_feast-0.0.1/setup.cfg
E       Creating tar archive
E       removing 'my_feast-0.0.1' (and everything under it)
E       Failed to lock ArtifactURL(raw_url='https://files.pythonhosted.org/packages/69/2b/f0ee898c3270d965300ec30b0bf06e062c4cc92f35d17ae6046f429c5067/SQLAlchemy-1.4.25.tar.gz#sha256=1adf3d25e2e33afbcd48cfad8076f9378793be43e7fec3e4334306cac6bec138', normalized_url='https://files.pythonhosted.org/packages/69/2b/f0ee898c3270d965300ec30b0bf06e062c4cc92f35d17ae6046f429c5067/SQLAlchemy-1.4.25.tar.gz', scheme='https', path='/packages/69/2b/f0ee898c3270d965300ec30b0bf06e062c4cc92f35d17ae6046f429c5067/SQLAlchemy-1.4.25.tar.gz'). Could not obtain its content for analysis.
...
```

This exactly mirrors the original report.

Reply:
Unfortunately its a very slow test, ~50s on my network. I spent a good long while trying to come up with a less expensive repro case though and this replacement of `feast` with `my_feast` was the best I could do - shaved ~15s.